c7n/resources/vpc.py

```diff
@@ -215,11 +216,11 @@ def process(self, resources, event=None):
         fattrs = list(sorted(self.attrs.intersection(self.data.keys())))
         self.ports = 'Ports' in self.data and self.data['Ports'] or ()
         self.only_ports = 'OnlyPorts' in self.data and self.data['OnlyPorts'] or ()
+        self.max_cidr_size = 'IpRanges' in self.data and self.data['IpRanges']['value'] or ()
         for f in fattrs:
             fv = self.data.get(f)
             if isinstance(fv, dict):
-                if 'key' not in fv:
-                    fv['key'] = f
+                fv['key'] = f
```

Comment:
incidentally the additionally conditions here also extract nicely to their own methods, handle ports, and handle cidrs. aka try this https://gist.github.com/kapilt/86bc759d16ac70e8c2ee05a36cbd0f7e